adafruit_ssd1306.py

```diff
@@ -48,20 +48,33 @@ class _SSD1306(framebuf.FrameBuffer):
     """Base class for SSD1306 display driver"""
 
     # pylint: disable-msg=too-many-arguments
-    def __init__(self, buffer, width, height, *, external_vcc, reset):
+    def __init__(self, buffer, width, height, *, external_vcc, reset, page_addressing):
```

Comment:
For backwards compatibility:

```suggestion
    def __init__(self, buffer, width, height, *, external_vcc, reset, page_addressing=False):
```

Reply:
Thanks @tannewt for reviewing the changes and the addition for backwards compatibility.

The parameter `page_addressing` is set to False in the child classes `SSD1306_I2C()` and `SSD1306_SPI()` and 
I am not sure it is a good idea to set page_addressing=False here in the parent class `_SSD1306` also?  I followed the behaviour of the optional parameters `reset` and `external_vcc`, where the defaults are set in these main child classes (and reset becomes a non-optional parameter).

With the `_` prefix on the parent class `_SSD1306` I am not sure it was ever intended (or has been used in the past) as an entry point?  I happy to set `page_addressing` to False here also if you're sure it does not affect / break inheritance to the child classes?

Reply:
On always using _page addressing mode_ instead -- this is an option for sure.
Some points on this:
1. **(negative)** _Horizontal addressing mode_ has some advantages in that it is a little more efficient and potential faster, since data for the whole screen is sent all at once.  There is a little overhead in the library too in order to break up the data into pages and send.  Realistically I do not think this is a huge deal.
2. **(negative)** Although the logic I have added to split the data into the correct number of pages and of the right length should work perfectly on all screens according to the specs, I have only tested on 64x32 and 128x32 screens, so this new addition could do with some more thorough testing if it was made the default. 
3. **(positive)** Most libraries -- including those that often accompany the screens as demonstration codes -- use the simpler, more widely accepted _page addressing mode_. 
4. **(negative)** Lastly, this library was designed for the Adafruit displays, which as far as I know use the more advanced chipset SSD1306, so understand it makes sense to use the (potentially better) extended command set, including _horizontal addressing_.

Reply:
> I'm happy to set `page_addressing` to False here also if you're sure it does not affect / break inheritance to the child classes?

It's good as-is. I missed that the child classes set it.

> On always using _page addressing mode_ instead -- this is an option for sure.

I think it's safer to do what you are doing to leave it with the same default behavior.